### ISSUE TITLE
Setting "min" should not automatically set "value" to same as "min" if value was empty

### ISSUE THREAD

**Issue by Haprog** [vaadin/vaadin-time-picker#123]
Currently time picker seems to have this logic that if there is a `value` set and then `min` or `max` is updated so that the old value would be out of valid range, it automatically adjusts the value in this case to clamp it between min/max (If old value was bigger than max, it would set the value to max. If old value was lower than min, it would set the value to min).

But currently also when no `value` has been set (`value === ''`, when you set `min` it automatically sets `value` to be the same as `min`. This is clearly a bug as empty value is still valid too.

Reproducible at https://cdn.vaadin.com/vaadin-time-picker/2.0.4/demo/#time-picker-basic-demos if you inspect the first time picker and set some min for it:
```js
timePicker.min = '08:00'; // See that 'value' also gets set to '08:00' automatically.
```

You can also see in the "Time Range" demo which has `min="09:00"` that it also has a value of `"09:00"` even though the value hasn't been explicitly set.

**Comment by martinisraelsen** [comment]
Concur with @Haprog. Just ran into this issue today. I tried manually clearing the value with time picker.setValue() but this does not clear the value.

**Comment by Haprog** [comment]
On the web component you can at least clear the value manually just by setting `timePicker.value = ""` (after min has been set). I guess it might be that when using the Java API it doesn't synchronize the value back to server side when the value has been automatically changed by setting min. That might not be a problem anymore after this is fixed though.

From the Java API (as a workaround for now) you should be able to clear it after setting min by clearing the value with JS like:
```java
timePicker.getElement().executeJs("this.value = '';");
```

**Comment by Haprog** [comment]
This has been fixed in `vaadin-time-picker` v2.0.5 ([GitHub](https://github.com/vaadin/vaadin-time-picker/releases/tag/v2.0.5), [npm](https://www.npmjs.com/package/@vaadin/vaadin-time-picker/v/2.0.5), [Maven/WebJar](https://repo1.maven.org/maven2/org/webjars/bowergithub/vaadin/vaadin-time-picker/2.0.5/)).
